### PR TITLE
postgres[minor]: add postgres checkpoint implementation

### PR DESCRIPTION
**Pull request by eyurtsev**
Adds checkpoint implementation using psycopg

### CODE REVIEW

File: libs/partners/postgres/langchain_postgres/checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+                    {
+                        "thread_id": thread_id,
+                        "thread_ts": checkpoint["ts"],
+                        "parent_ts": parent_ts if parent_ts else None,
+                        "checkpoint": pickle.dumps(checkpoint),
+                    },
+                )
+
+        return {
+            **config,  # TODO(Nuno): Confirm
```
Comment (eyurtsev) [inline libs/partners/postgres/langchain_postgres/checkpoint.py:308]: @nfcampos need review here -- unclear what is meant to be returned by this method or why

File: libs/partners/postgres/langchain_postgres/checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+                    checkpoint BYTEA NOT NULL,
+                    thread_ts TIMESTAMPTZ NOT NULL,
+                    parent_ts TIMESTAMPTZ,
+                    PRIMARY KEY (thread_id, thread_ts)
+                );
+                """
+            )
+
+    @staticmethod
+    async def adrop_table(connection: psycopg.AsyncConnection, /) -> None:
```
Comment (eyurtsev) [inline libs/partners/postgres/langchain_postgres/checkpoint.py:158]: might rename to adrop_schema to match `acreate_schema`

File: libs/partners/postgres/langchain_postgres/checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    This is a reference implementation of a checkpoint saver for Postgres.
+
+    The implementation provides some functionality to create the necessary table
+    and schema for the checkpoint saver, as well as to get and put checkpoints.
+
+    Examples:
+
+        .. code-block:: python
+
+            from langchain_core.messages import AIMessage, HumanMessage, SystemMessage
```
Comment (eyurtsev) [inline libs/partners/postgres/langchain_postgres/checkpoint.py:65]: need to update here

File: libs/partners/postgres/langchain_postgres/checkpoint.py [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+                    )
+                    value = await cur.fetchone()
+                    if value:
+                        return CheckpointTuple(
+                            config,
+                            pickle.loads(value[0]),
+                            {
+                                "configurable": {
+                                    "thread_id": thread_id,
+                                    "thread_ts": value[1].isoformat(),
```
Comment (eyurtsev) [inline libs/partners/postgres/langchain_postgres/checkpoint.py:237]: forcing to isoformat()

File: libs/partners/postgres/langchain_postgres/checkpoint.py [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+                await cur.execute(
+                    """
+                    INSERT INTO checkpoints (thread_id, thread_ts, parent_ts, checkpoint)
+                    VALUES (%(thread_id)s, %(thread_ts)s, %(parent_ts)s, %(checkpoint)s)
+                    ON CONFLICT (thread_id, thread_ts) 
+                    DO UPDATE SET checkpoint = EXCLUDED.checkpoint;
+                    """,
+                    {
+                        "thread_id": thread_id,
+                        "thread_ts": checkpoint["ts"],
```
Comment (eyurtsev) [inline libs/partners/postgres/langchain_postgres/checkpoint.py:301]: dropped validation with isoformat which was present with opengpts

File: libs/partners/postgres/langchain_postgres/checkpoint.py (hunk 1 above)
Comment (nfcampos) [reply libs/partners/postgres/langchain_postgres/checkpoint.py:308]: this is the config that uniquely identifies the checkpoint that was just saved (the primary key)

**Pull request merged**